fairscale/optim/oss.py

```diff
@@ -334,29 +368,38 @@ def local_state_dict(self) -> dict:
         """
         return self.optim.state_dict()
 
-    def state_dict(self) -> Dict[str, Any]:
+    def state_dict(self, all_ranks: bool = False) -> Dict[str, Any]:
```

Comment:
why all this: make it possible to get the state on all ranks. In that case a prealable "consolidate_state_dict" is not needed anymore